### ISSUE TITLE
Error on "current_heating_setpoint" of BHT-002-GCLZB (TS0601 - _TZE204_5toc8efa)

### ISSUE THREAD

**Issue by Jakmaster199** [Koenkk/zigbee-herdsman-converters#6838]
Hi,

The "current_heating_setpoint" doesn't work properly on "_TZE204_5toc8efa".

First of all, it should change in 0.5 degree steps. By other side, when you set from the own device 14.0 degrees it reports 140 on Z2M and if you set 14.0 degrees on Z2M it sets to 0 on the device.
![image](https://github.com/Koenkk/zigbee-herdsman-converters/assets/1685032/2a560854-bd68-45d9-a213-c0d4b2fe7ab3)

Also, on HA is wrong too since it passes the readed value:
![image](https://github.com/Koenkk/zigbee-herdsman-converters/assets/1685032/730a648e-e638-4858-8fe0-351e7e65023d)

Thanks!

**Comment by Koenkk** [comment]
This is probably due to https://github.com/Koenkk/zigbee-herdsman-converters/pull/6581 (@DasRed), can you both provide the data/database.db entries of your devices?

**Comment by Jakmaster199** [comment]
Yep,

The database entry of" _TZE204_5toc8efa" is this one:
```json
{
    "id": 2,
    "type": "Router",
    "ieeeAddr": "0xa4c13887ec4eb066",
    "nwkAddr": 41241,
    "manufId": 4417,
    "manufName": "_TZE204_5toc8efa",
    "powerSource": "Mains (single phase)",
    "modelId": "TS0601",
    "epList": [
        1,
        242
    ],
    "endpoints": {
        "1": {
            "profId": 260,
            "epId": 1,
            "devId": 81,
            "inClusterList": [
                4,
                5,
                61184,
                0
            ],
            "outClusterList": [
                25,
                10
            ],
            "clusters": {
                "genBasic": {
                    "attributes": {
                        "65503": "�@'-g�@'-g",
                        "65506": 56,
                        "65508": 0,
                        "modelId": "TS0601",
                        "manufacturerName": "_TZE204_5toc8efa",
                        "powerSource": 1,
                        "zclVersion": 3,
                        "appVersion": 74,
                        "stackVersion": 0,
                        "hwVersion": 1,
                        "dateCode": ""
                    }
                }
            },
            "binds": [],
            "configuredReportings": [],
            "meta": {}
        },
        "242": {
            "profId": 41440,
            "epId": 242,
            "devId": 97,
            "inClusterList": [],
            "outClusterList": [
                33
            ],
            "clusters": {},
            "binds": [],
            "configuredReportings": [],
            "meta": {}
        }
    },
    "appVersion": 74,
    "stackVersion": 0,
    "hwVersion": 1,
    "dateCode": "",
    "zclVersion": 3,
    "interviewCompleted": true,
    "meta": {},
    "lastSeen": 1704236101659,
    "defaultSendRequestWhen": "immediate"
}
```

Also, full database is attached here -> [database.zip](https://github.com/Koenkk/zigbee-herdsman-converters/files/13842675/database.zip) :)

**Comment by DasRed** [comment]
Hey, 

here are the database entries. I have 3 of these devices and add all 3 of them. I hope, this helps.

BTW Everything works fine if I set the "current heating setpoint" value via Zigbee2mqtt UI or via device interface


```json
{
    "id": 15,
    "type": "Router",
    "ieeeAddr": "0xa4c138e643063f2f",
    "nwkAddr": 20442,
    "manufId": 4417,
    "manufName": "_TZE204_5toc8efa",
    "powerSource": "Mains (single phase)",
    "modelId": "TS0601",
    "epList": [
        1,
        242
    ],
    "endpoints": {
        "1": {
            "profId": 260,
            "epId": 1,
            "devId": 81,
            "inClusterList": [
                4,
                5,
                61184,
                0
            ],
            "outClusterList": [
                25,
                10
            ],
            "clusters": {
                "genBasic": {
                    "attributes": {
                        "65503": "�M)-i�M)-i",
                        "65506": 56,
                        "65508": 0,
                        "stackVersion": 0,
                        "dateCode": "",
                        "appVersion": 74
                    }
                }
            },
            "binds": [],
            "configuredReportings": [],
            "meta": {}
        },
        "242": {
            "profId": 41440,
            "epId": 242,
            "devId": 97,
            "inClusterList": [],
            "outClusterList": [
                33
            ],
            "clusters": {},
            "binds": [],
            "configuredReportings": [],
            "meta": {}
        }
    },
    "appVersion": 74,
    "stackVersion": 0,
    "hwVersion": 1,
    "dateCode": "",
    "zclVersion": 3,
    "interviewCompleted": true,
    "meta": {},
    "lastSeen": 1704461117367,
    "defaultSendRequestWhen": "immediate"
}
```

```json
{
    "id": 38,
    "type": "Router",
    "ieeeAddr": "0xa4c138fa29a4ee69",
    "nwkAddr": 29278,
    "manufId": 4417,
    "manufName": "_TZE204_5toc8efa",
    "powerSource": "Mains (single phase)",
    "modelId": "TS0601",
    "epList": [
        1,
        242
    ],
    "endpoints": {
        "1": {
            "profId": 260,
            "epId": 1,
            "devId": 81,
            "inClusterList": [
                4,
                5,
                61184,
                0
            ],
            "outClusterList": [
                25,
                10
            ],
            "clusters": {
                "genBasic": {
                    "attributes": {
                        "65503": "f�'-i",
                        "65506": 56,
                        "65508": 0,
                        "stackVersion": 0,
                        "dateCode": "",
                        "appVersion": 74
                    }
                }
            },
            "binds": [],
            "configuredReportings": [],
            "meta": {}
        },
        "242": {
            "profId": 41440,
            "epId": 242,
            "devId": 97,
            "inClusterList": [],
            "outClusterList": [
                33
            ],
            "clusters": {},
            "binds": [],
            "configuredReportings": [],
            "meta": {}
        }
    },
    "appVersion": 74,
    "stackVersion": 0,
    "hwVersion": 1,
    "dateCode": "",
    "zclVersion": 3,
    "interviewCompleted": true,
    "meta": {},
    "lastSeen": 1704461120557,
    "defaultSendRequestWhen": "immediate"
}
```


```json
{
    "id": 39,
    "type": "Router",
    "ieeeAddr": "0xa4c1380e556638d7",
    "nwkAddr": 38688,
    "manufId": 4417,
    "manufName": "_TZE204_5toc8efa",
    "powerSource": "Mains (single phase)",
    "modelId": "TS0601",
    "epList": [
        1,
        242
    ],
    "endpoints": {
        "1": {
            "profId": 260,
            "epId": 1,
            "devId": 81,
            "inClusterList": [
                4,
                5,
                61184,
                0
            ],
            "outClusterList": [
                25,
                10
            ],
            "clusters": {
                "genBasic": {
                    "attributes": {
                        "65503": "ƶ*-i",
                        "65506": 56,
                        "65508": 0,
                        "modelId": "TS0601",
                        "manufacturerName": "_TZE204_5toc8efa",
                        "powerSource": 1,
                        "zclVersion": 3,
                        "appVersion": 74,
                        "stackVersion": 0,
                        "hwVersion": 1,
                        "dateCode": ""
                    }
                }
            },
            "binds": [],
            "configuredReportings": [],
            "meta": {}
        },
        "242": {
            "profId": 41440,
            "epId": 242,
            "devId": 97,
            "inClusterList": [],
            "outClusterList": [
                33
            ],
            "clusters": {},
            "binds": [],
            "configuredReportings": [],
            "meta": {}
        }
    },
    "appVersion": 74,
    "stackVersion": 0,
    "hwVersion": 1,
    "dateCode": "",
    "zclVersion": 3,
    "interviewCompleted": true,
    "meta": {},
    "lastSeen": 1704461118458,
    "defaultSendRequestWhen": "immediate"
}
```

**Comment by Koenkk** [comment]
@Jakmaster199 does this issue also happen with the `local_temperature` and `max_temperature` values?

**Comment by Jakmaster199** [comment]
Not, `local_temperature` and `max_temperature` seems to be fine:
```
    "local_temperature": 21.5,
    "local_temperature_calibration": -4,
    "max_temperature": 350,
    "max_temperature_limit": 350,
    "min_temperature_limit": 50,
```

Can be related to anything about the manufacturer or model?
My thermostat is from BSEED and the model is GL86HTBZ1 but get recognized as MOES with model "BHT-002-GCLZB".

**Comment by Koenkk** [comment]
`max_temperature` is 350, is that fine?

> Can be related to anything about the manufacturer or model?

Yes, but the problem is that that all properties (modelId/hwVersion/manufacturerName/dateCode) are all the same so we cannot distinguish them in the code.

**Comment by Jakmaster199** [comment]
No, 350º is not good... I didn't saw that there's no decimals on number.

```
    "current_heating_setpoint": 180,
    "deadzone_temperature": 50,
    "local_temperature": 17.5,
    "local_temperature_calibration": -4,
    "max_temperature": 350,
    "max_temperature_limit": 350,
    "min_temperature_limit": 50,
```

Should be like this:

```
    "current_heating_setpoint": 18,
    "deadzone_temperature": 5.0,
    "local_temperature": 17.5,
    "local_temperature_calibration": -0.4,
    "max_temperature": 35.0,
    "max_temperature_limit": 35.0,
    "min_temperature_limit": 5.0,
```

So seems that all temperatures are wrong readed/setted except the `local_temperature`.

Maybe the easiest method to fix that is to add an option to "scale" the temperatures like the power meters "specific settings" where you can modify precision and calibration.

**Comment by Koenkk** [comment]
It's a bit complicated to solve this cleanly from Z2M, and I don't know if other users will also run into this. Maybe the easiest thing to do now is that you stop z2m, change the `manufacturerName` for this device in your `data/database.db` to `_TZE200_aoclfnxz` and start it. That should fix all the problems including the 0.5 step size.
